### ISSUE TITLE
Yambo 4.5.3: Unexpected behaviour of --enable-dp

### ISSUE THREAD

**Issue by klust** [yambo-code/yambo#31]
According to the help information of configure, ``--enable-dp=no`` should be equivalent to ``--disable-dp``, or, given that the default is to configure for single precision, not specifying this option. This however does not work and is a bug in the option handling for ``--enable-dp``, where only action is taken if nothing is specified or if the value is yes, leaving the variable ``build_precision`` uninitialized (and causing the build of PETSc to fail). The actual bug is in ``config/yambo_specific.m4`` (Yambo 4.5.3) or ``config/m4/yambo_specific.m4`` (GitHub sources found on 3 February 2021).  The problem is on lines 66-68 which currently reads:
```bash
def_dp=""
if test x"$enable_dp" = "x";    then enable_dp="no";     build_precision="single"; fi
if test x"$enable_dp" = "xyes"; then def_dp="-D_DOUBLE"; build_precision="double"; fi
```
which should become, e.g., 
```bash
if test x"$enable_dp" = "x";    then enable_dp="no"; fi
def_dp=""
build_precision="single"
if test x"$enable_dp" = "xyes"; then def_dp="-D_DOUBLE"; build_precision="double"; fi
```
(and the first line is only needed if you want to ensure that ``enable_dp`` is not empty which may help avoiding trouble further down the configure script.)

The workaround is of course simply, never use ``--disable-dp`` or ``--enable-dp=no``, but it is confusing and one needs to be an expert software installer to realise that the bug during the build process is due to specifying a single precision build in this way (which is perfectly valid according to ``./configure --help``).

**Comment by sangallidavide** [comment]
Check if this solution looks reasonable to you

```
def_dp=""
build_precision=""
if test x"$enable_dp" = "x";    then enable_dp="no"; fi
if test x"$enable_dp" = "xno" ; then def_dp="";          build_precision="single"; fi
if test x"$enable_dp" = "xyes"; then def_dp="-D_DOUBLE"; build_precision="double"; fi
if test x"$build_precision" = "x"; then AC_MSG_ERROR([Non correct specification of compilation precision]); fi
```

P.S.: sorry for the long time it took to reply

**Comment by sangallidavide** [comment]
Fixed in bug-fixes branch with commit [e5bafe9](https://github.com/yambo-code/yambo/commit/e5bafe9e01d7a179b520628b85afcdbab58af074)
